ompi/mca/mtl/psm2/mtl_psm2_component.c

```diff
@@ -387,6 +372,26 @@ ompi_mtl_psm2_component_init(bool enable_progress_threads,
         ompi_mtl_psm2_set_shadow_env (ompi_mtl_psm2_shadow_variables + i);
     }
 
+#if OPAL_CUDA_SUPPORT
+    /*
+     * If using CUDA enabled OpenMPI, the user likely intends to
+     * run with CUDA buffers. So, force-set the envvar here if user failed
+     * to set it.
+     */
+    cuda_env = getenv("PSM2_CUDA");
+    if (!cuda_env) {
```

Comment:
I'll repeat what I said in the last PR: don't you want to check and see if there are CUDA devices present before you do this?  Just being compiled for CUDA support is not any kind of guarantee that there are CUDA devices present.

Indeed, if someone uses an Open MPI compiled with CUDA support (e.g., via OpenHPC) in an environment without CUDA devices, it will be quite confusing to them as to why they are getting help messages about CUDA support (because it won't matter at all).

Run-time sensing what devices are present is a big theme of Open MPI.

Reply:
Ok, I'll add some CUDA device detection here. Can I simply issue  `lspci -nn | grep -i "10de"`

and check the return value? Any other suggestions?

Reply:
No, that will not scale (imagine N processes on a server all executing that at once).

Reply:
We literally *just* disabled hwloc CUDA detection in master (c341b5347509037c145f0c3b8fd990d1511e6908) per https://github.com/open-mpi/ompi/pull/4257#issuecomment-332900393.  @sjeaugey is this a good reason to turn it back on?

Reply:
Added some CUDA device detection logic in the updated patch.